tests/jobs/scala/src/main/scala/MultiConfs.scala

```diff
@@ -0,0 +1,16 @@
+import org.apache.spark.SparkConf
+
+/**
+  * Application that outputs the Spark configurations.
+  * Specifically we want to test whether long strings of multiple arg=values are passed through correctly.
+  * E.g. 
+  * dcos spark run --submit-args="--conf=spark.driver.extraJavaOptions='val1 val2 val3' --class MultiConfs"
+  */ 
+object MultiConfs {
+    def main(args: Array[String]): Unit = {
+        println("Running MultiConfs app. Printing out all config values:")
+        val APPNAME = "MultiConfs App"
+        val conf = new SparkConf().setAppName(APPNAME)
+        conf.getAll.foreach(println)
```

Comment:
One additional check that tests the whole thing end-to-end: pass in `-Dparam1=\"valA valB\"` as one of the extraJavaOptions, and then inside this app, you can check the value of System.getProperty("param1") to make sure it was set correctly.

Reply:
Good idea, I'll update.